### PR TITLE
Navbar typescript redone

### PR DESCRIPTION
**Pull request by EDsCODE**
**Describe the Pull Request**
- modifies navbar.js (re PR of #191) to use hooks and typescriptified

A clear and concise description of the features implemented in the PR of bugs fixed.

**Todo**

~~- [ ] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request~~ (refactor)
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver

### CODE REVIEW

**Comment by EDsCODE** [comment]
QA'd navbar changing tabs and orgs work. 

Not quite sure how to interpret codecov issue. Also, I setup using the new quick setup script and it works great. However, the dashboard page is handing (perpetual loading). May have been noted but making an observation here just in case.

**Comment by tristanhcole** [comment]
> QA'd navbar changing tabs and orgs work.
Great! I'll take a look today.

> Not quite sure how to interpret codecov issue. Also, I setup using the new quick setup script and it works great. However, the dashboard page is handing (perpetual loading). May have been noted but making an observation here just in case.
Yeah, seems to be a caching issue in regards to perpetual loading, raised in #dev-chat slack channel.

File: app/client/components/navBar/OrgSwitcher.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+          display: switcherActive ? "block" : "none",
+          zIndex: 99
+        }}
+      >
+        <DropdownContentTitle>Switch Organisation</DropdownContentTitle>
+        {orgs.map(org => {
+          return (
+            <DropdownContentText key={org.id} onClick={() => selectOrg(org.id)}>
+              {org.name}
+            </DropdownContentText>
```
Comment (tristanhcole) [inline app/client/components/navBar/OrgSwitcher.tsx:89]: nit: type `org: Organisation` should be added

File: app/client/components/navBar/OrgSwitcher.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+const OrgSwitcher: React.FunctionComponent<Props> = props => {
+  const [switcherActive, setSwitcherActive] = React.useState(false);
+
+  const login: LoginState = useSelector((state: ReduxState) => state.login);
+  let { email } = login;
+  const organisations: Organisation[] = useSelector((state: ReduxState) => Object.keys(state.organisations.byId).map(
+    id => state.organisations.byId[Number(id)]
+  ));
+  const activeOrganisation = useSelector((state: ReduxState) => state.organisations.byId[Number(state.login.organisationId)]);
```
Comment (tristanhcole) [inline app/client/components/navBar/OrgSwitcher.tsx:23]: Haven't used hooks before! Cool!

File: app/client/nav.jsx [hunk 3]
```diff
@@ -232,7 +232,7 @@ const LoadingSpinnerWrapperSideBarActive = () => {
 const PageWrapper = ({ noNav, component: Component, ...props }) => {
   return (
     <ErrorBoundary>
-      {noNav ? null : <NavBar location={props.location} />}
```
Comment (tristanhcole) [inline app/client/nav.jsx:235]: Definitely need to keep cleaning up unused props

**Pull request merged**